metagraph/src/common/sorted_sets/sorted_set_disk_base.cpp

```diff
@@ -122,15 +120,16 @@ void SortedSetDiskBase<T>::dump_to_file(bool is_done) {
         encoder.add(v);
     }
     total_chunk_size_bytes_ += encoder.finish();
+    chunk_count_++;
+
     data_.resize(0);
+
     if (is_done) {
         async_merge_l1_.remove_waiting_tasks();
```

Comment:
this is a bit confusing, which tasks do we expect to still be waiting if `is_done` is true?

Reply:
The "merge_l1" tasks. If we are about to merge all the chunks anyway, there is little point to do this l1 pre-merge blocking the ultimate merge. This is a minor optimization.